packages/jest-runner/src/index.ts

```diff
@@ -164,11 +217,25 @@ class TestRunner {
           },
           globalConfig: this._globalConfig,
           path: test.path,
-        });
+        }) as PromiseWithCustomMessage<TestResult>;
+
+        if (promise.UNSTABLE_onCustomMessage) {
+          // TODO: Get appropriate type for `onCustomMessage`
```

Comment:
can we do this?

Reply:
I tried tinkering with this, but resulted in not very clean implementations. Moreover it is linked to typings for `sendMessageToJest`, so maybe we can tackle this when we get a better solution for `sendMessageToJest`?

Reply:
sounds good!